chore(exp): Add docs for new experiments and libs (#353)
litmuschaos/litmus-docs: docs/node-io-stress.md

- Ensure that the `node-io-stress` experiment resource is available in the cluster  by executing                         `kubectl get chaosexperiments` in the desired namespace. If not, install from [here](https://hub.litmuschaos.io/api/chaos/1.8.0?file=charts/generic/node-io-stress/experiment.yaml)

## Entry Criteria

- Application pods are healthy on the respective Nodes before chaos injection

## Exit Criteria

- Application pods may or may not be healthy post chaos injection

## Details

- This experiment causes disk stress on the Kubernetes node. The experiment aims to verify the resiliency of applications that share this disk resource for ephemeral or persistent storage purposes.
- The amount of disk stress can be either specifed as the size in percentage of the total free space on the file system or simply in Gigabytes(GB). When provided both it will execute with the utilization percentage specified and non of them are provided it will execute with default value of 10%.

> - Application implies services. Can be reframed as:

Reviewer: This is not needed

Tests application resiliency upon replica evictions caused due IO stress on the available Disk space.

## Integrations

> - Node IO Stress can be effected using the chaos library: `litmus`

Reviewer: injected

> - The desired chaos library can be selected by setting `litmus` as value for the env variable `LIB` 

Reviewer: Since we only support litmus lib, this is not needed

## Steps to Execute the Chaos Experiment

- This Chaos Experiment can be triggered by creating a ChaosEngine resource on the cluster. To understand the values to provide in a ChaosEngine specification, refer [Getting Started](getstarted.md/#prepare-chaosengine)

- Follow the steps in the sections below to create the chaosServiceAccount, prepare the ChaosEngine & execute the experiment.

### Prepare chaosServiceAccount

> - Use this sample RBAC manifest to create a chaosServiceAccount in the desired (app) namespace. This example consists of the minimum necessary role permissions to execute the experiment.

Reviewer: Lets use `codeBlocks` with camelCase wherever the CR spec attributes are discussed

kind: ClusterRole
metadata:
  name: node-io-stress-sa
  labels:
    name: node-io-stress-sa
rules:
- apiGroups: ["","litmuschaos.io","batch","apps"]
  resources: ["pods","jobs","pods/log","events","chaosengines","chaosexperiments","chaosresults"]
  verbs: ["create","list","get","patch","update","delete"]
- apiGroups: [""]
  resources: ["nodes"]
  verbs: ["get","list"]
---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  name: node-io-stress-sa
  labels:
    name: node-io-stress-sa
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: node-io-stress-sa
subjects:
- kind: ServiceAccount
  name: node-io-stress-sa
  namespace: default
```

### Prepare ChaosEngine

- Provide the application info in `spec.appinfo`
- Provide the auxiliary applications info (ns & labels) in `spec.auxiliaryAppInfo`
- Override the experiment tunables if desired in `experiments.spec.components.env`
- To understand the values to provided in a ChaosEngine specification, refer [ChaosEngine Concepts](chaosengine-concepts.md)

#### Supported Experiment Tunables

<table>
  <tr>
    <th> Variables </th>
    <th> Description  </th>

>     <th> Specify In ChaosEngine </th>

Reviewer: ```suggestion
    <th> Specify in ChaosEngine </th>
```

    <th> Notes </th>
  </tr>
  <tr>
    <td> TOTAL_CHAOS_DURATION </td>

>     <td> The time duration for chaos insertion (seconds) </td>

Reviewer: ```suggestion
    <td> The time duration for chaos (seconds) </td>
```

    <td> Optional </td>
    <td> Defaults to 120 </td>
  </tr>
  <tr>
    <td> FILESYSTEM_UTILIZATION_PERCENTAGE </td>
    <td> Specify the size as percentage of free space on the file system  </td>
    <td> Optional  </td>

>     <td> DEfault to 10%</td>

Reviewer 2: Default?

  </tr>
  <tr>
    <td> FILESYSTEM_UTILIZATION_BYTES </td>

>     <td> Specify the size in GigaBytes(GB) for the IO disk stress. While providing this keep <code>FILESYSTEM_UTILIZATION_PERCENTAGE<code> empty owtherwise the experiment runs with the percentage value </td>

Reviewer: ```suggestion
    <td> Specify the size in GigaBytes(GB).  <code>FILESYSTEM_UTILIZATION_PERCENTAGE</code> & <code>FILESYSTEM_UTILIZATION_BYTES</code> are mutually exclusive. If both are provided, <code>FILESYSTEM_UTILIZATION_PERCENTAGE</code> is prioritized</td>
```

    <td> Optional  </td>

>     <td> No default value </td>

Reviewer 2: can we less it blank or nil?

  </tr>
  <tr>
    <td> NUMBER_OF_WORKERS </td>
    <td> It is the number of IO workers involved in IO disk stress </td>
    <td> Optional  </td>
    <td> Defaults to 4 </td>
  </tr>   
  <tr>
    <td> APP_NODE </td>
    <td> Name of the node subjected to IO stress </td>
    <td> Optional  </td>
    <td> If not provided. It will select the app node from appinfo randomly</td>
  </tr>  
   <tr>
    <td> LIB  </td>
    <td> The chaos lib used to inject the chaos </td>
    <td> Optional </td>
    <td> Defaults to <code>litmus</code> </td>
  </tr>
   <tr>
    <td> LIB_IMAGE  </td>
    <td> Image used to run the stress command </td>
    <td> Optional  </td>

>     <td> Defaults to <code>litmuschaos/litmus-go:latest<code> </td>

Reviewer 2: go-runner?

    appkind: 'deployment'
  chaosServiceAccount: node-io-stress-sa
  monitoring: false
  # It can be delete/retain
  jobCleanUpPolicy: 'delete'
  experiments:
    - name: node-io-stress
      spec:
        components:
          env:
            # set chaos duration (in sec) as desired
            - name: TOTAL_CHAOS_DURATION
              value: '120'

            ## specify the size as percentage of free space on the file system
            - name: FILESYSTEM_UTILIZATION_PERCENTAGE
              value: '10'
            
             ## enter the name of the desired node
            - name: APP_NODE
              value: ''
```

### Create the ChaosEngine Resource

- Create the ChaosEngine manifest prepared in the previous step to trigger the Chaos.

  `kubectl apply -f chaosengine.yml`

- If the chaos experiment is not executed, refer to the [troubleshooting](https://docs.litmuschaos.io/docs/faq-troubleshooting/) 
  section to identify the root cause and fix the issues.

### Watch Chaos progress

- View the status of the pods as they are subjected to IO disk stress. 

  `watch -n 1 kubectl get pods -n <application-namespace>`

- Get inside the application pod

>   `kubectl exec -it <application-pod-name> -n <application-namespace> -- bash`

Reviewer: This can be done outside the pod as well, i.e., on the node console

- Monitor the capacity filled up on the host filesystem

>   `Watch du -h`

Reviewer: ```suggestion
  `watch du -h`
```

  
### Abort/Restart the Chaos Experiment

> - To stop the pod-delete experiment immediately, either delete the ChaosEngine resource or execute the following command:

Reviewer 2: pod-delete?
